### ISSUE TITLE
Long Lists / HLists wrapped as a staircase

### ISSUE THREAD

**Issue by johnynek** [scalameta/scalafmt#843]
## Problem

Scalafmt formats code like this when using a long (24 element) HList
```scala
val foo = Call(
  firstItem ::
    secondItem ::
      thirdItem ::
        fourthItem ::
          HNil)
```

## Expectation

I would like the formatted output to look like this:
```scala
val foo = Call(
  firstItem ::
    secondItem ::
    thirdItem ::
    fourthItem ::
    HNil)
```

**Comment by olafurpg** [comment]
What version are you on? I think this got fixed in 0.6

**Comment by johnynek** [comment]
0.5.2-RC1. I'll try again with 0.6.

**Comment by pjrt** [comment]
Closing since this was fixed in 0.6. Reopen if there's an issue.
